x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -243,12 +244,14 @@ const getWaterfallItems = (items: TraceAPIResponse['trace']['items']) =>
  * @param waterfallItems
  */
 const reparentSpans = (waterfallItems: IWaterfallItem[]) => {
-  return waterfallItems.map(waterfallItem => {
+  // clones the waterfall items to avoid mutating the original
+  const clonedWaterfallItems = cloneDeep(waterfallItems);
```

Comment:
What was the issue with mutating (which I agree should be avoided)?

Reply:
There was no issue, I only want to make it a pure function with no side effects on external variables.